react-client/src/components/pages/root/root.tsx

```diff
@@ -0,0 +1,228 @@
+import * as React from 'react';
+import AppBar from '@mui/material/AppBar';
+import Box from '@mui/material/Box';
+import Toolbar from '@mui/material/Toolbar';
+import IconButton from '@mui/material/IconButton';
+import Typography from '@mui/material/Typography';
+import Menu from '@mui/material/Menu';
+import MenuIcon from '@mui/icons-material/Menu';
+import Container from '@mui/material/Container';
+
+import Button from '@mui/material/Button';
+
+import MenuItem from '@mui/material/MenuItem';
+import AdbIcon from '@mui/icons-material/Adb';
+import { useEffect, useState } from "react"
+import { Outlet, useNavigate } from 'react-router-dom';
+
+import { useAuth, useUserContext } from '../../../context';
+import { jwtDecode } from 'jwt-decode';
+const pages = ['home', 'vacations', 'about us'];
+const settings = ['Profile', 'Account', 'Dashboard', 'Logout'];
+
+
+
+
+
+function ResponsiveAppBar() {
+
+  
+
+  const navigate = useNavigate();
+  const { token ,setToken} = useAuth(); 
+  const { fullName  } = useUserContext();
+  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
+  const [anchorElNav, setAnchorElNav] = useState<null | HTMLElement>(null);
+  const [anchorElUser, setAnchorElUser] = useState<null | HTMLElement>(null);
+  
+  useEffect(() => {
+    const checkTokenExpiration = () => {
+      if (!token) {
+        setIsLoggedIn(false);
+        return;
+      }
+
+      try {
+        const decoded: any = jwtDecode(token);
+        const isExpired = decoded.exp * 1000 < Date.now();
```

Comment:
its not accepted to check the expiration in the client
it usually in the backend, its not an operation that is recommended to do in the client side